staging/src/k8s.io/apiserver/pkg/server/options/encryptionconfig/config.go

```diff
@@ -0,0 +1,116 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package encryptionconfig
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apiserver/pkg/storage/value"
+	"k8s.io/apiserver/pkg/storage/value/encrypt/identity"
+)
+
+// TransformerConfig is a blanket interface for all provider configurations
+type TransformerConfig interface {
+	// GetPrefixTransformer constructs and returns a PrefixTransformer using the provided configuration
+	GetPrefixTransformer() (value.PrefixTransformer, error)
+	// IsNil abstracts the nil check for the interface, since nil pointer checks on interface don't work
```

Comment:
This shouldn't be necessary.  You really just want to iterate over a concrete config and then return a prefix transformer - I doubt this interface is necessary.